core/templates/dev/head/pages/exploration-editor-page/services/learner-answer-details-data.service.ts

```diff
@@ -0,0 +1,108 @@
+require('domain/utilities/UrlInterpolationService.ts');
+require('domain/statistics/LearnerAnswerInfoObjectFactory.ts');
+require('domain/statistics/LearnerAnswerDetailsObjectFactory.ts');
+
+angular.module('oppia').factory('LearnerAnswerDetailsDataService', [
+  '$http', '$q', 'ExplorationDataService', 'LearnerAnswerDetailsObjectFactory',
+  'LearnerAnswerInfoObjectFactory', 'UrlInterpolationService',
+  function(
+      $http, $q, ExplorationDataService, LearnerAnswerDetailsObjectFactory,
+      LearnerAnswerInfoObjectFactory, UrlInterpolationService) {
+    var _expId = ExplorationDataService.explorationId;
+    var _data = [];
+    var learnerAnswerInfoData = null;
+    var LEARNER_ANSWER_INFO_DATA_URL = (
+      '/learneranswerinfohandler/learner_answer_details/<entity_type>' +
+      '/<entity_id>');
+
+    var _fetchLearnerAnswerInfoData = function(
+        successCallback, errorCallback) {
+      var learnerAnswerInfoDataUrl = UrlInterpolationService.interpolateUrl(
+        LEARNER_ANSWER_INFO_DATA_URL, {
+          entity_type: 'exploration',
+          entity_id: _expId});
+
+      $http.get(learnerAnswerInfoDataUrl).then(function(response) {
+        learnerAnswerInfoData = angular.copy(
+          response.data.learner_answer_info_dict_list);
+        for (var i = 0; i < learnerAnswerInfoData.length; i++) {
+          var stateName = Object.keys(learnerAnswerInfoData[i])[0];
+          var learnerAnswerInfoDicts = learnerAnswerInfoData[i][stateName];
+          var learnerAnswerDetails = (
+            LearnerAnswerDetailsObjectFactory.createDefaultLearnerAnswerDetails(
+              _expId, stateName, learnerAnswerInfoDicts.map(
+                LearnerAnswerInfoObjectFactory.createFromBackendDict)));
+          _data.push(learnerAnswerDetails);
+        }
+        if (successCallback) {
+          successCallback(response.data);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    var _deleteLearnerAnswerInfo = function(
+        entityId, stateName, learnerAnswerInfoId, successCallback,
+        errorCallback) {
+      var learnerAnswerInfoDataUrl = UrlInterpolationService.interpolateUrl(
+        LEARNER_ANSWER_INFO_DATA_URL, {
+          entity_type: 'exploration',
+          entity_id: entityId});
+      $http['delete'](learnerAnswerInfoDataUrl, {
+        params: {
+          state_name: stateName,
+          learner_answer_info_id: learnerAnswerInfoId
+        }
+      }).then(function(response) {
+        if (successCallback) {
+          successCallback(response.status);
+        }
+      }, function(errorResponse) {
+        if (errorCallback) {
+          errorCallback(errorResponse.data);
+        }
+      });
+    };
+
+    return {
+      getData: function() {
+        return _data;
+      },
+      fetchLearnerAnswerInfoData: function() {
+        return $q(function(resolve, reject) {
```

Comment:
Now you can just do:

```javascript
return _fetchLearnerAnswerInfoData();
```

Same for `_deleteLearnerAnswerInfo`.

Reply:
Done

Reply:
Done